addons/mail/static/src/models/emoji_text_field_view.js

```diff
@@ -0,0 +1,55 @@
+/** @odoo-module **/
+
+import { attr, clear, one, Model} from '@mail/model';
+
+
+Model({
+    name: 'EmojiTextFieldView',
+    recordMethods: {
+        /**
+         * On selecting an emoji from the list
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickEmoji(ev) {
+            const emojiCharacter = ev.currentTarget.dataset.codepoints;
+
+            const [start, end] = this.textSelection;
+            const originalContent = this.textInputRef.el.value;
+            const left = originalContent.slice(0, start);
+            const right = originalContent.slice(end, originalContent.length);
+            this.textInputRef.el.value = left + emojiCharacter + right;
+
+            // trigger onInput from input_field hook to set field as dirty
+            this.textInputRef.el.dispatchEvent(new InputEvent('input'));
+            // trigger change event to trigger the model onChange method
+            this.textInputRef.el.dispatchEvent(new Event('change'));
```

Comment:
==============================

Actually I'm not sure why the issue we last discussed came up but it seems to work fine with "keydown" and "input" events alone in the end. I don't actually need "onchange" to prevent owl from overwriting the value, just "commitChanges".

So no issue, we can do as you said.

Reply:
This seemed like the test would get fairly involved so I removed the onchange test I had added. It's more a test for something like a tour than a unit test, probably.

But let me know if you think a test would be worth it.